tests/functional/encryption/test_encryption_keyrotation.py

```diff
@@ -344,3 +421,200 @@ def test_osd_keyrotation_with_kms(self, multi_pvc_factory, pod_factory):
 
         # Disable Keyrotation
         self.keyrotation.disable_keyrotation()
+
+
+@green_squad
+@encryption_at_rest_required
+@vault_kms_deployment_required
+@skipif_external_mode
+class TestNoobaaKeyrotationWithKMS:
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        """
+        Resetting the default value of KeyRotation
+        """
+
+        def finalizer():
+            kr_obj = KeyRotation()
+            kr_obj.set_keyrotation_defaults()
+
+        request.addfinalizer(finalizer)
+
+    @retry(UnexpectedBehaviour, tries=10, delay=20)
+    def compare_noobaa_old_keys_with_new_keys(
+        self, noobaa_keyrotation, old_noobaa_backend_key, old_noobaa_volume_key
+    ):
+        """
+        Compare noobaa old keys with new keys.
+        args:
+            noobaa_keyrotation: obj: NoobaaKeyrotation object
+            old_noobaa_volume_key: str: old noobaa_volume_key
+
+        """
+        (
+            new_noobaa_backend_key,
+            new_noobaa_backend_secret,
+        ) = noobaa_keyrotation.get_noobaa_backend_secret(kms_deployment=True)
+        (
+            new_noobaa_volume_key,
+            new_noobaa_volume_secret,
+        ) = noobaa_keyrotation.get_noobaa_volume_secret()
+
+        if new_noobaa_backend_key == old_noobaa_backend_key:
+            raise UnexpectedBehaviour("Noobaa Key Rotation is not happend")
+
+        if new_noobaa_volume_key == old_noobaa_volume_key:
+            raise UnexpectedBehaviour("Noobaa Key Rotation is not happend.")
+
+        log.info(
+            f"Noobaa Backend key rotated {new_noobaa_backend_key} : {new_noobaa_backend_secret}"
+        )
+        log.info(
+            f"Noobaa Volume key rotated {new_noobaa_volume_key} : {new_noobaa_volume_secret}"
+        )
```

Comment:
It seems like this code is repeated 2 times. Please remove one block of code and made it accessible for both the tests